src/fontra/views/editor/scene-controller.js

```diff
@@ -1333,6 +1396,32 @@ function getSelectedContours(path, pointSelection) {
   return [...selectedContours];
 }
 
+function getSelectedOpenContours(
+  path,
+  pointSelection,
+  ignoreSinglePointContour = true
+) {
+  if (!path || !pointSelection) {
+    return [];
+  }
+  const selectedContours = new Set();
+  for (const contourIndex of getSelectedContours(path, pointSelection)) {
+    if (path.contourInfo[contourIndex].isClosed) {
+      // skip if contour is closed already
+      continue;
+    }
+    const contour = path.getContour(contourIndex);
+    const numOnCourvePoints = contour.pointTypes.filter((x) => x == 0).length;
```

Comment:
It is wasteful to build an array and then only use it for its length. You can do this with `.reduce()` instead.

Typo: `numOnCourvePoints` -> `numOnCurvePoints`

Reply:
thanks and thanks. done.